frontend/views/containers/chatroom/poll-message-content/VoterAvatars.vue

```diff
@@ -0,0 +1,251 @@
+<template lang='pug'>
+.c-voter-avatars(
+  v-if='voters.length'
+  @click='showTooltip'
+  v-on-clickaway='closeTooltip'
+)
+  template(v-for='entry in votersToDisplay')
+    .c-number-avatar(v-if='isNum(entry)') +{{ entry }}
+    avatar-user.c-user-avatar(v-else :key='entry' :username='entry' size='xs')
+
+  .c-voters-tooltip(
+    :class='{ "is-active": ephemeral.isTooltipActive }'
+  )
+    .c-voters-tooltip-overlay(@click.stop='closeTooltip')
+
+    .c-voters-tooltip-content
+      header.c-voters-tooltip-header
+        i18n.is-title-4.c-tooltip-title.has-ellipsis(
+          tag='h2'
+          :args='{ option: optionName }'
+        ) Voted for “{option}”
+
+        modal-close.c-tooltip-close-btn(@close='closeTooltip')
+
+      ul.c-voters-list
+        li.c-voter-item(v-for='(votername, index) in voters' :key='votername + index')
+          avatar-user.c-voter-item-avatar(:username='votername' size='xs')
+          span.c-voter-item-name {{ getDisplayName(votername) }}
+</template>
+
+<script>
+import { mapGetters } from 'vuex'
+import { mixin as clickaway } from 'vue-clickaway'
+import AvatarUser from '@components/AvatarUser.vue'
+import ModalClose from '@components/modal/ModalClose.vue'
+
+export default ({
+  name: 'VoterAvatars',
+  mixins: [
+    clickaway
+  ],
+  components: {
+    AvatarUser,
+    ModalClose
+  },
+  data () {
+    return {
+      ephemeral: {
+        isTooltipActive: false
+      }
+    }
+  },
+  props: {
+    voters: Array,
+    optionName: String
+  },
+  computed: {
+    ...mapGetters([
+      'globalProfile'
+    ]),
+    votersToDisplay () {
+      const restNum = this.voters.length - 2
+
+      return [
+        restNum > 0 && restNum,
+        ...this.voters.slice(0, 2)
+      ].filter(Boolean)
+    }
+  },
+  methods: {
+    isNum (val) {
+      return typeof val === 'number'
+    },
+    showTooltip () {
+      if (!this.ephemeral.isTooltipActive) {
+        this.ephemeral.isTooltipActive = true
+      }
+    },
+    closeTooltip () {
+      if (this.ephemeral.isTooltipActive) {
+        this.ephemeral.isTooltipActive = false
+      }
+    },
```

Comment:
If I were you, I will remove the `if` statements and reduce the LOC, and create only one function `setToolTip` instead of two functions.

This is fine too. Anyway, it's up to you.

Reply:
@Silver-IT That's a simpler way but we are using `v-on-clickaway` 3rd-party directive here and this requires a single component method to be passed as the directive argument. Also, specifying an inline arrow function like `() => setTooltipActive(false)` leads to a console error too. will remove the `if` conditionals btw.